### ISSUE TITLE
Document expected structure of bag storage

### ISSUE THREAD

**Issue by NikolausDemmel** [bosch-ros-pkg/bagbunker#116]
I don't think it is currently documented how the `/mnt/bags` directory is expected to be structured, or did I miss it.

I guess it should say that bag files are searched in all subfolders and that `.md5` files are expected and how to create them.

What actually happens when two bags with the same name in different folders are found?

**Comment by pjreed** [comment]
I just tried to start up an instance of bagbunker, and when I tell it to scan for bags, I get output like this:
```
$ bagbunker scan /mnt/bags
2016-07-25 17:02:05,551 WARNING bagbunker.scanner Discarding incomplete fileset [(2, FileInfo(dirpath='/mnt/bags', name='fixed-reordered-migrated-pwm-data_2015-09-02-10-25-13_2.bag'))]
2016-07-25 17:02:05,552 WARNING marv.scanner ('MissingMD5', '/mnt/bags', '2013-03-14-16-28-15-reorder.orig.bag')
2016-07-25 17:02:05,552 WARNING marv.scanner ('MissingMD5', '/mnt/bags', '2014-07-25-15-37-22.bag')
2016-07-25 17:02:05,552 WARNING marv.scanner ('MissingMD5', '/mnt/bags', '2014-07-25-15-37-22.orig.bag')
```

I assume that's related to this issue?  Is the expected structure documented anywhere?

**Comment by NikolausDemmel** [comment]
I don't think it is documented, but you need a checksum file for each bag file beforehand, which you can generate, e.g. with:

```
md5sum foo.bag > foo.bag.md5
```

**Comment by pjreed** [comment]
Thanks, things are looking better now.  It successfully scanned most of my bag files, but I have several  files that have warning messages that look like so:
```
2016-07-25 21:45:58,171 WARNING marv.storage skipped MD5 mismatch <Fileset "encoder_calibration_data_2015-09-03-11-24-45" type="bag" dir="/mnt/bags" files=1 md5="23f3f8c5cd11279f591d5a14b5b1690e">
2016-07-25 21:45:58,181 WARNING marv.storage skipped MD5 mismatch <Fileset "encoder_calibration_data_2015-09-04-12-56-56" type="bag" dir="/mnt/bags" files=1 md5="3406da578b4711fc2c89c3efac348d2a">
```

I've double-checked the contents of their .md5 files, and they seem correct to me.

**Comment by dejanpan** [comment]
Hm. That is weird. Could you maybe upload these 2 bags somewhere for me? I
will give a try tomorrow.

**Comment by pjreed** [comment]
[encoder_calibration_bags.zip](https://github.com/bosch-ros-pkg/bagbunker/files/382555/encoder_calibration_bags.zip)

They're fairly small, so I'll just attach them here.

**Comment by NikolausDemmel** [comment]
This seems to be a permission info with your files. Try this to verify:

```
docker exec -it bbdev bash -c "cd /mnt/bags && md5sum -c *.md5"
```

As a fix you could make them world-readable:

```
chmod o+r *.bag
```

Indeed the error message should be improved. Bagbunker simply calls `md5sum -c ...` in a subprocess and reports the error message you see if the return code is not 0, regardless of what exactly is the cause for the non-zero exit code.

**Comment by pjreed** [comment]
That fixed some of them, but not all -- it turns out I also had an issue with some of my .md5 files where the file name in them was referenced relative to the top-level bag directory when they should have been relative to the directory containing the bag.

For example, I had this
```
$ cat /mnt/bags/encoder_calibration_data/encoder_calibration_data_2015-09-02-16-49-05.bag.md5 
567594deb03e963a561452fbdc9acd4c  ./encoder_calibration_data/encoder_calibration_data_2015-09-02-16-49-05.bag
```

Changing it to this fixed the issue:
```
567594deb03e963a561452fbdc9acd4c  ./encoder_calibration_data_2015-09-02-16-49-05.bag
```

**Comment by NikolausDemmel** [comment]
> it turns out I also had an issue with some of my .md5 files where the file name in them was referenced relative to the top-level bag directory when they should have been relative to the directory containing the bag.

Good point, that should also make it into the documentation. I had a similar issue initially with absolute paths in the md5sum files. The paths were for the host and did not match the absolute path inside the container. Relative paths, i.e. `./bagfilename.bag` is the way to go.
